osu.Game/Overlays/UserProfileOverlay.cs

```diff
@@ -143,24 +170,27 @@ public void ShowUser(IUser user, IRulesetInfo? ruleset = null)
 
             sectionsContainer.ScrollToTop();
 
+            if (!API.IsLoggedIn)
+                return;
```

Comment:
Note that `IsLoggedIn` will return `true` whilst the user is authenticating, which means the `loadingLayer` will be shown - however if they fail to authenticate on this overlay it will not be hidden again (the request will fail). Adding an additional check here to make it return when it's `Connecting` fixes this, however not sure if that's the best way to handle it -
```suggestion
            if (!API.IsLoggedIn || API.State.Value == APIState.Connecting)
                return;
```

Reply:
If this is even a real issue (which I have doubts about - what does "fail to authenticate" mean?) it should not need to be handled locally in such a way and should be handled at a higher level.

Reply:
I have changed this check to be `if (API.State.Value != APIState.Online)` as there are other places where this is done before sending a request, eg. https://github.com/ppy/osu/blob/d2380bd8401491d13d70bf0122ce1df341f8620e/osu.Game/Beatmaps/BeatmapUpdaterMetadataLookup.cs#L78 The main valid flow where a user will fail to authenticate is incorrect credentials, and `IAPIProvider.IsLoggedIn` will return true during the time where they attempt the login to when the auth request fails - my fault for misunderstanding the implementation on this one.